code/game/objects/items/grenades/grenade.dm

```diff
@@ -68,9 +68,6 @@
 	log_grenade(user, T) //Inbuilt admin procs already handle null users
 	if(user)
 		add_fingerprint(user)
-		if(iscarbon(user))
-			var/mob/living/carbon/C = user
-			C.throw_mode_on()

```

Comment:
put a line above this:
`if(!in_throw_mode)`
then make dropping/changing hands end throw mode

Reply:
```suggestion
			if(!C.in_throw_mode)
			    C.throw_mode_on()
```

Reply:
kind of pointless to do that since he still needs to undelete all these lines which needs its own commit anyway but OK